### ISSUE TITLE
Benchmarks

### ISSUE THREAD

**Issue by prisae** [simpeg/simpeg#733]
As SimPEG is approaching version 1.0 I was wondering if benchmarking the most important bits would be a good idea, now that the API is fairly stable?

I started a few moths ago with empymod using airspeed-velocity ([asv](https://github.com/airspeed-velocity/asv)), and am very pleased with it. It showed me unexpected bottlenecks and how seemingly tiny, unimportant commits had a serious impact (positive or negative) on speed.

The main maintainer of `asv` at the moment is Pauli Virtanen, BDFL of scipy. As such it is used for the benchmark-suites of numpy and scipy, and therefore ideally suited for scientific packages such as SimPEG. It is very actively maintained, and Pauli responds very quick to issues. (Examples of asv: [numpy](https://pv.github.io/numpy-bench), [scipy](https://pv.github.io/scipy-bench), [empymod](http://empymod.github.io/asv/).)

I post it here in SimPEG, but it probably would also apply to discretize, geoana, and others.

**Comment by prisae** [comment]
The only small downside of asv is that there is no free online service like travis-ci, coveralls, etc. You have to run it on your own. But I think it wouldn't be too difficult to make on the UBC server a cronjob that runs `asv run NEW` and subsequently `asv publish; asv gh-pages`, which would then daily run all new commits to master, and publish the result to the defined location.

**Comment by lheagy** [comment]
👍 This would be great to get in shortly after the simulation refactor #786!

**Comment by prisae** [comment]
I agree. Ping me afterwards if it is getting forgotten. It is something that will grow gradually. We just have to set it up for a few basic functions - Once it is in place it can grow over time, and should particularly so for new things added.
